src/Nox.Types/Types/DateTime/DateTime.cs

```diff
@@ -0,0 +1,198 @@
+    {
+        if (!System.DateTime.TryParse(datetime, out System.DateTime dateTime))
+        {
+            throw new ArgumentOutOfRangeException(nameof(datetime), datetime, "Invalid datetime");
+        }
+
+        return new DateTime
+        {
+            Value = dateTime,
+        };
+    }
+
+    private static bool IsValidDate(int year, int month, int day, int hour = 0, int minute = 0, int second = 0)
+    {
+        try
+        {
+            _ = new System.DateTime(year, month, day, hour, minute, second);
+            return true;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    public static DateTime operator +(DateTime dt, TimeSpan ts)
+    {
+        System.DateTime newDateTime = dt.Value.Add(ts);
+        return DateTime.From(newDateTime);
+    }
+
+    public static TimeSpan operator -(DateTime dateTime1, DateTime dateTime2)
+    {
+        return dateTime1.Value - dateTime2.Value;
+    }
+
+    public static bool operator >(DateTime dateTime1, DateTime dateTime2)
+    {
+        return dateTime1.Value > dateTime2.Value;
+    }
+
+    public static bool operator <(DateTime dateTime1, DateTime dateTime2)
+    {
+        return dateTime1.Value < dateTime2.Value;
+    }
+
+    public static bool operator ==(DateTime dateTime1, DateTime dateTime2)
+    {
+        if (dateTime1 is null && dateTime2 is null)
+            return true;
+
+        if (dateTime1 is not null)
+            return dateTime1.Equals(dateTime2);
+
+        return false;
+    }
+
+    public static bool operator !=(DateTime dateTime1, DateTime dateTime2)
+    {
+        return !(dateTime1 == dateTime2);
+    }
+
+    public static bool operator <=(DateTime dateTime1, DateTime dateTime2)
+    {
+        return dateTime1.Value <= dateTime2.Value;
+    }
+
+    public static bool operator >=(DateTime dateTime1, DateTime dateTime2)
+    {
+        return dateTime1.Value >= dateTime2.Value;
+    }
+
+    public override string ToString()
+    {
+        return Value.ToString(CultureInfo.InvariantCulture);
+    }
+
+    public string ToString(CultureInfo cultureInfo)
+    {
+        return Value.ToString(cultureInfo);
+    }
+
+    public string ToString(string format, CultureInfo cultureInfo)
+    {
+        return Value.ToString(format, cultureInfo);
+    }
+
+    public string ToString(string format)
+    => Value.ToString(format, CultureInfo.InvariantCulture);
+
+    public string ToString(IFormatProvider formatProvider)
+    {
+        return Value.ToString(formatProvider);
+    }
+
+    public string ToString(string format, IFormatProvider formatProvider)
+    {
+        return Value.ToString(format, formatProvider);
+    }
+
+    public DateTime Add(TimeSpan ts)
+    {
+        Value = Value.Add(ts);
+        return this;
+    }
+
+    public override bool Equals(object obj)
+    {
+        return obj is DateTime time &&
+               base.Equals(obj) &&
+               Value == time.Value &&
```

Comment:
The Value underlinetype is Datetime, so doing a equal on the value should be enough:

` return obj is DateTime time &&
               base.Equals(obj) && Value.Equals(time.Value) `